gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ResolveProductDependenciesTask.java

```diff
@@ -254,93 +259,56 @@ final void computeProductDependencies() throws IOException {
-                                            .getEffectiveManifest()
-                                            .getAttributes()
-                                            .get(RecommendedProductDependencies.SLS_RECOMMENDED_PRODUCT_DEPS_KEY))
-                                    .map(Object::toString);
-                        }
-                    } else {
-                        if (!artifact.getFile().exists()) {
-                            log.debug("Artifact did not exist: {}", artifact.getFile());
-                            return Stream.empty();
-                        } else if (!com.google.common.io.Files.getFileExtension(
-                                        artifact.getFile().getName())
-                                .equals("jar")) {
-                            log.debug("Artifact is not jar: {}", artifact.getFile());
-                            return Stream.empty();
-                        }
-
-                        Manifest manifest;
-                        try {
-                            ZipFile zipFile = new ZipFile(artifact.getFile());
-                            ZipEntry manifestEntry = zipFile.getEntry("META-INF/MANIFEST.MF");
-                            if (manifestEntry == null) {
-                                log.debug("Manifest file does not exist in jar for '${coord}'");
-                                return Stream.empty();
-                            }
-                            manifest = new Manifest(zipFile.getInputStream(manifestEntry));
-                        } catch (IOException e) {
-                            log.warn(
-                                    "IOException encountered when processing artifact '{}', file '{}', {}",
-                                    artifactName,
-                                    artifact.getFile(),
-                                    e);
-                            return Stream.empty();
-                        }
+        return artifactStream
+                .map(ResolveProductDependenciesTask::getProductDependenciesFromArtifact)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .flatMap(Collection::stream)
+                .distinct()
+                .collect(Collectors.toList());
+    }
 
-                        pdeps = Optional.ofNullable(manifest.getMainAttributes()
-                                .getValue(RecommendedProductDependencies.SLS_RECOMMENDED_PRODUCT_DEPS_KEY));
-                    }
-                    if (!pdeps.isPresent()) {
-                        log.debug(
-                                "No product dependency found for artifact '{}', file '{}'",
-                                artifactName,
-                                artifact.getFile());
-                        return Stream.empty();
-                    }
+    static Optional<Collection<ProductDependency>> getProductDependenciesFromArtifact(ResolvedArtifact artifact) {
```

Comment:
Why not return an empty collection instead of `Optional.empty()`? That would simplify your stream logic above.

Reply:
doh.  yes, that is cleaner.  That's what I get for reusing code and not looking closely at mods.